extensions/notebook/src/book/bookTocManager.ts

```diff
@@ -20,13 +20,19 @@ export interface IBookTocManager {
 	createBook(bookContentPath: string, contentFolder: string): Promise<void>;
 	addNewTocEntry(pathDetails: TocEntryPathHandler, bookItem: BookTreeItem, isSection?: boolean): Promise<void>;
 	recovery(): Promise<void>;
+	movedFiles: Map<string, string>;
+	tocStates: Map<string, tocState>;
 	enableDnd: boolean;
 }
 
 export interface quickPickResults {
 	quickPickSection?: vscode.QuickPickItem,
 	book?: BookTreeItem
 }
+export interface tocState {
+	current: string,
```

Comment:
Comments please, what are `current` and `previous` storing in this case? The entire toc yaml? Any concerns with storing many copies of the same (potentially large) string?

Reply:
I refactored this logic.  I stored the toc JupyterBookSections as an object, and there's no longer current/previous states.